contracts/src/v0.8/llo-feeds/v0.4.0/test/verifier/DestinationVerifierTestRewardsMultiVefifierFeeManager.t.sol

```diff
@@ -0,0 +1,144 @@
+import {DestinationVerifier} from "../../../v0.4.0/DestinationVerifier.sol";
+import {DestinationVerifierProxy} from "../../../v0.4.0/DestinationVerifierProxy.sol";
+import {AccessControllerInterface} from "../../../../shared/interfaces/AccessControllerInterface.sol";
+import {Common} from "../../../libraries/Common.sol";
+
+
+contract MultiVerifierBillingTests is MultipleVerifierWithMultipleFeeManagers {
+
+
+
+  uint8 MINIMAL_FAULT_TOLERANCE = 2;
+  address internal constant DEFAULT_RECIPIENT_1 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_1"))));
+  address internal constant DEFAULT_RECIPIENT_2 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_2"))));
+  address internal constant DEFAULT_RECIPIENT_3 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_3"))));
+  address internal constant DEFAULT_RECIPIENT_4 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_4"))));
+  address internal constant DEFAULT_RECIPIENT_5 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_5"))));
+  address internal constant DEFAULT_RECIPIENT_6 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_6"))));
+  address internal constant DEFAULT_RECIPIENT_7 = address(uint160(uint256(keccak256("DEFAULT_RECIPIENT_7"))));
+
+
+  bytes32[3] internal s_reportContext;
+  V3Report internal s_testReport;
+
+  function setUp() public virtual override {
+    MultipleVerifierWithMultipleFeeManagers.setUp();
+    s_reportContext[0] = bytes32(abi.encode(uint32(5), uint8(1)));
+    s_testReport = generateReportAtTimestamp(block.timestamp);
+  }
+
+  function _verify(DestinationVerifierProxy proxy, bytes memory payload, address feeAddress, uint256 wrappedNativeValue, address sender) internal {
+    address originalAddr = msg.sender;
+    changePrank(sender);
+
+    proxy.verify{value: wrappedNativeValue}(payload, abi.encode(feeAddress));
+
+    changePrank(originalAddr);
+  }
+
+  function generateReportAtTimestamp(uint256 timestamp) public pure returns (V3Report memory) {
+    return
+      V3Report({
+        feedId: FEED_ID_V3,
+        observationsTimestamp: OBSERVATIONS_TIMESTAMP,
+        validFromTimestamp: uint32(timestamp),
+        nativeFee: uint192(DEFAULT_REPORT_NATIVE_FEE),
+        linkFee: uint192(DEFAULT_REPORT_LINK_FEE),
+        // ask michael about this expires at, is it usually set at what blocks
+        expiresAt: uint32(timestamp) + 500,
+        benchmarkPrice: MEDIAN,
+        bid: BID,
+        ask: ASK
+      });
+  }
+
+  function payRecipients(bytes32 poolId, address[] memory recipients, address sender) public {
+    //record the current address and switch to the recipient
+    address originalAddr = msg.sender;
+    changePrank(sender);
+
+    //pay the recipients
+    rewardManager.payRecipients(poolId, recipients);
+
+    //change back to the original address
+    changePrank(originalAddr);
+  }
+
+  function test_multipleFeeManagersAndVerifiers() public {
```

Comment:
@Fletch153  one note here, a call to `setConfig` ends up in a call in reward manager to `setRewardRecipients`.  The implication is that we can't set the same DONConfigID rewards recipient from two different verifier contracts. If we do, we get a `invalidPoolId` as that DONConfigID already has some weights set in.

in code, the following will fail:

```solidity
verifier1 = DestinationVerrifier(..);
verifier2 = DestinationVerrifier(..);
feeManager.addVerifier(address(verifier1));
feeManager2.addVerifier(address(verifier2));
rewardManager.addFeeManager(feeManager);
rewardManager.addFeeManager(feeManager2);

verifier1.setConfig(signerAddrs, FAULT_TOLERANCE, weights);
// this will fail with invalidPoolId
verifier2.setConfig(signerAddrs, FAULT_TOLERANCE, weights);

```

Reply:
This is fine, in this case rewardrecipients will be an empty array